src/vs/platform/terminal/node/terminalProfiles.ts

```diff
@@ -115,7 +130,7 @@ async function detectAvailableWindowsProfiles(configuredProfilesOnly: boolean, f
 	return resultProfiles;
 }
 
-async function transformToTerminalProfiles(entries: IterableIterator<[string, ITerminalProfileObject]>, defaultProfileName: string | undefined, fsProvider: IFsProvider, logService?: ILogService, variableResolver?: ExtHostVariableResolverService, workspaceFolder?: IWorkspaceFolder): Promise<ITerminalProfile[]> {
+async function transformToTerminalProfiles(entries: IterableIterator<[string, ITerminalProfileObject]>, defaultProfileName: string | undefined, fsProvider: IFsProvider, logService?: ILogService, variableResolver?: (text: string[]) => Promise<string[]>, workspaceFolder?: IWorkspaceFolder): Promise<ITerminalProfile[]> {
```

Comment:
might benefit from horizontal layout like the arguments for `detectAvailableWindowsProfiles`